cm-vault/README.md

```diff
@@ -0,0 +1,968 @@
+# Deploy and Setup Cert-manager and Vault
+
+## Goals
+
+1. Deploy and setup Cert-manager and Vault in test environment.
+
+2. Automate to configure Vault PKI Secrets Engine and Vault authentication.
+
+3. Automate to create a ClusterIssuer that represents the certificate authority Vault.
+
+## Dependencies Graph
```

Comment:
Suggestion: add a short explanation what are these dependencies i.e `Terragrunt is used to ensure the correct order of dependency installation, i.e that cert-manager `ClusterIssuer` is not created before installing cert-manager. See dependency graph below:`